spec/game_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'rspec'
+require_relative '../lib/req.rb'
+
+RSpec.describe Game do
+  let(:board) { Board.new('Eli', 'Satt') }
+  let(:g) { Game.new('Eli', 'X', 'Satt', 'O') }
+  let(:player1) { Player.new('Eli', 'X') }
+  let(:player2) { Player.new('Satt', 'O') }
+
+  describe '#check_winner' do
+    it 'winner_update must be false since the moves_counter is lesser than 5' do
+      expect(g.winner_update(player1, board)).to eq(false)
+    end
+  end
+
+  describe '#new_game?' do
+    it 'new_game must return true if user press [Q]' do
+      g.stub(:gets).and_return("q\n")
+      expect(g.new_game?).to eq(true)
+    end
+  end
+
+  describe '#new_game?' do
+    it 'new_game must return true if user press [Enter]' do
+      g.stub(:gets).and_return("\n")
+      expect(g.new_game?).to eq(false)
+    end
+  end
+
+  describe '#make_your_move?' do
+    it 'make_your_move must return value stored in @moves_counter' do
+      player1.stub(:gets).and_return("1\n")
```

Comment:
Please using this way is actually manual, please apply similarly to tie update.

Reply:
Hello Zlatan! Thank you for your guidance.

How can we avoid using stub:gets if the path that follows our method require the user input?

1st -- game calls #make_your_move
2nd -- #make_your move calls #move (that is in Player class)
3rd -- when it's done the #make_your_move returns @moves_counter

___________________________________________________


#make_your_move (line 97 in Game class)
![Screenshot 2019-04-03 at 1 21 40 PM](https://user-images.githubusercontent.com/1456388/55503329-100af900-5614-11e9-8174-435def4ec561.png)


#move (line 13 in Game class)
![Screenshot 2019-04-03 at 1 22 03 PM](https://user-images.githubusercontent.com/1456388/55503328-100af900-5614-11e9-80db-d09d4f7ca08a.png)

Our new test case launches the console asking for the user input
![Screenshot 2019-04-03 at 1 22 31 PM](https://user-images.githubusercontent.com/1456388/55503327-100af900-5614-11e9-90cb-43d495ed9795.png)

and finally when we run our tests the method ask for the user input.
![Screenshot 2019-04-03 at 1 23 11 PM](https://user-images.githubusercontent.com/1456388/55503325-100af900-5614-11e9-81bd-c2384b6ce1da.png)

So, we don't know how to exactly proceed in this.
Can you help us?